lib/cloud/aws/policy_statements.go

```diff
@@ -433,3 +433,48 @@ func StatementAccessGraphAWSSync() *Statement {
 		Resources: allResources,
 	}
 }
+
+// StatementForAWSIdentityCenterAccess returns AWS IAM policy statement that grants
+// permissions required for Teleport identity center client.
+// TODO(sshah): make the roles more granular by restricting resources scoped to
+// partivular AWS identity center region+arn.
```

Comment:
I think to properly scope these permissions will require very careful conditional permissions, because it grants `iam:CreateRole` and `iam:AttachRolePolicy` which can be used to elevate permissions. Also note that any teleport discovery service will be able to get a token to assume the IdP integration role, so it may be time to think about scoping the trust policy further.

Reply:
Definitely. This is a hard TODO I set and I will resolve it before release. Trent is coming back from vacation next week and I plan to sync with him to figure out the final permissions (the provisioning service is mostly done but we have some more rough edges to scope down) and conditions we can apply here. 

The gist is: most of the actions will be scoped to the identity center region + ARN we ask from user. The role create and attach policies are risky but required since we provision AWS role on demand when access request is granted in Teleport or when user is enlisted to an access list. During test, I found there are some actions that require organization-wide resource scope so I want to properly take time and plan for them, hence it is currently loosely defined to address all the permissions we know we require for the identity center integration.

Happy to add some minimum constraints in this PR if you think that is still important before we merge to master.

Reply:
ah ok, I'm fine with the permissions for now since it doesn't do anything yet.

I am more concerned about the existing integration trust policy which allows for any teleport discovery service to assume the role though. I don't see mention of that in the RFD, so i'll take this comment to the RFD

Reply:
Is there some way we can prevent this from actually being exercised until we address the permission scope? Or at least force it to issue a warning if it gets hit? 

I'm just concerned in case we fill in the rest of the Identity Center work and forget to  revisit this.